### ISSUE TITLE
Unable to run test due to "Error [ERR_MODULE_NOT_FOUND]: Cannot find module *.mjs"

### ISSUE THREAD

**Issue by amoshydra** [jessealama/decimal128#94]
### Current

```
> decimal128@11.2.0 test
> jest

 FAIL  tests/subtract.test.js
  ● Test suite failed to run

    Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\projects\repository\github.com\jessealama\decimal128\src\decimal128.mjs' imported from D:\projects\repository\github.com\jessealama\decimal128\tests\subtract.test.js



 FAIL  tests/rational.test.js
  ● Test suite failed to run

    Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'D:\projects\repository\github.com\jessealama\decimal128\src\rational.mjs' imported from D:\projects\repository\github.com\jessealama\decimal128\tests\rational.test.js


 // ... repeat 12 more times

Test Suites: 14 failed, 14 total
Tests:       0 total
Snapshots:   0 total
Time:        0.434 s
Ran all test suites.
```

### Expected
To be able to test

## Reproduce

checkout to today's latest
```sh
git checkout 66a0909717894af09b3c613d77668f329dfceaf4 
npm i
```

```sh
npm test
```

**Comment by amoshydra** [comment]
I manage to resolve the issue to a certain extend,
first - rename the tests to import `mts` (the real file extension) instead of the transpiled one

Then:
- https://www.npmjs.com/package/jest-light-runner#using-custom-nodejs-esm-loaders - on using custom loader
- https://github.com/nodejs/node/issues/51196 - on how to resolve the warning 

> to a certain extend

I said to a certain extends because coverage still doesn't quite work yet


Here is a PR that fix the issue with running `npm test` - https://github.com/jessealama/decimal128/pull/95

**Comment by amoshydra** [comment]
I am able to fix the coverage by using ts-jest following their guide on ESM Support in https://kulshekhar.github.io/ts-jest/docs/guides/esm-support

I have created another PR at #96

Let me know what you think

**Comment by amoshydra** [comment]
After inspecting [`.github/workflows/push.yaml`](https://github.com/jessealama/decimal128/blob/main/.github/workflows/push.yaml#L25), I come to realise that `npm test` and `npm run coverage` are meant to run after `npx tsc`.

Perhaps this is a no issue at all right now.

The original intention was actually to make test work for #93

**Comment by jessealama** [comment]
Right, I always compile the TypeScript before running tests. Is there a better way we could document that? Should we compile TypeScript as *part* of running the tests? That would ensure that tests always run on the compiled JS (and wouldn't even run if there were TypeScript compile errors), but it seems a bit heavy to me.

**Comment by amoshydra** [comment]
(1)
> . Is there a better way we could document that? 

usually it can be documented in the `README.md`, i.e.

`````md
### Running test

```sh
npx tsc
npm run test
```
``````

(2)
> That would ensure that tests always run on the compiled JS

This is reasonable.

In the projects I worked with, I generally write tests to import from the source code, not from transpiled code. 
My tests are also written in TypeScript and type checking is also done as part of the test.

In my opinion, testing against transpiled JS is considered much safer. Having the test fail due to TypeScript compile error is reasonable to me too.



(3) 
The more I think about it, this issue doesn't warrant a code change #96 #95.
My original motivation is also best resolved in #93 (*[updated](https://github.com/jessealama/decimal128/pull/93#issuecomment-2047790788)*) itself.

For this issue, I believe an update to the README should be sufficient.

**Comment by jessealama** [comment]
To be honest, I'd love to just write my tests in TypeScript, too. The main thing, in my opinion, the transpiled JS for end-users.

**Comment by jessealama** [comment]
Is this still an issue, given the changes in #121 ?

Also, FWIW: the filenames are `src/Decimal128.mts`, not `src/decimal128.mts`. Could case-(in)sensitivity be playing a role here?

**Comment by amoshydra** [comment]
oops, sorry for the late response.

> Could case-(in)sensitivity be paying a role here?

Nope, i was using a case-insensitive file system (Windows).


---

> Is this still an issue, given the changes in #121

I would say yes! but more of a documentation issue.

my last comment is still relevant.

> The more I think about it, this issue doesn't warrant a code change https://github.com/jessealama/decimal128/pull/96 https://github.com/jessealama/decimal128/pull/95.
My original motivation is also best resolved in https://github.com/jessealama/decimal128/pull/93 (https://github.com/jessealama/decimal128/pull/93#issuecomment-2047790788) itself.
>
> **For this issue, I believe an update to the README should be sufficient.**

This is from the perspective of a contributors when they first clone the project.

I'd assume most contributors will run `npm run test`, not knowing that they need to run `npm run build` first. Especially for project using `jest` or `vitest`.

---

let me know what you think.

I can help to:
- update the documentation to add a contributor section
- or we can use vitest to test directly on typescript code (vitest will internally transpile before testing)
